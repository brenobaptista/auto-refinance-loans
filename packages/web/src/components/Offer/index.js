import React from "react";
import styles from "./index.module.css";

const Offer = ({
  setOffer,
  monthlyPayment,
  newAnnualPercentage,
  newMonthlyPayment,
  newTotalPeriod,
}) => {
  return (
    <div className={styles.offer}>
      <label className={styles.loanInfo}>
        <span className={`${styles.discount} ${styles.textImportant}`}>
          Lower you payment by {(monthlyPayment - newMonthlyPayment).toFixed(2)}{" "}
          per month
        </span>
        <input
          type="radio"
          name="offer"
          onChange={() =>
            setOffer({
              newAnnualPercentage,
              newMonthlyPayment,
              newTotalPeriod,
            })
          }
          className={styles.radio}
        />
      </label>
      <div className={styles.loanInfo}>
        <div>APR</div>
        <div className={styles.textImportant}>{newAnnualPercentage * 100}%</div>
      </div>
      <hr />
      <div className={styles.loanInfo}>
        <div>Monthly payment</div>
        <div className={styles.textImportant}>${newMonthlyPayment}</div>
      </div>
      <hr />
      <div className={styles.loanInfo}>
        <div>Time remaining</div>
        <div className={styles.textImportant}>{newTotalPeriod} mo</div>
      </div>
    </div>
  );
};

export default Offer;
