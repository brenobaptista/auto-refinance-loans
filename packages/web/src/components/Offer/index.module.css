.offer {
  margin: 48px 0;
  font-size: 0.9rem;
  line-height: 1.5;
}

.loanInfo {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.discount {
  margin: 16px 0;
}

.radio {
  margin-left: 20px;
}

.textImportant {
  font-weight: 500;
  color: #000000;
}
