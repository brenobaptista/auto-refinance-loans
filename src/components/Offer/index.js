import React from "react";
import styles from "./index.module.css";

const Offer = ({
  discount,
  loanId,
  setLoanId,
  annualPercentage,
  monthlyPayment,
  timeRemaining,
}) => {
  return (
    <div className={styles.offer}>
      <label className={styles.loanInfo}>
        <span className={styles.discount}>
          Lower you payment by {discount} per month
        </span>
        <input
          type="radio"
          name="offer"
          value={loanId}
          onChange={(event) => setLoanId(event.target.value)}
          className={styles.radio}
        />
      </label>
      <div className={styles.loanInfo}>
        <div>APR</div>
        <div className={styles.textImportant}>{annualPercentage}</div>
      </div>
      <hr />
      <div className={styles.loanInfo}>
        <div>Monthly payment</div>
        <div className={styles.textImportant}>{monthlyPayment}</div>
      </div>
      <hr />
      <div className={styles.loanInfo}>
        <div>Time remaining</div>
        <div className={styles.textImportant}>{timeRemaining}</div>
      </div>
    </div>
  );
};

export default Offer;
