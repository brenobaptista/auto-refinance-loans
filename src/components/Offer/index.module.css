.offer {
  margin: 32px 0;
}

.loanInfo {
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.discount {
  margin: 16px 0;
  font-weight: 600;
  color: #000000;
}

.radio {
  margin-left: 15px;
}

.textImportant {
  font-weight: 600;
  color: #000000;
}
