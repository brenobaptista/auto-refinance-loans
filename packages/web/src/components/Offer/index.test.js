import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Offer from "./index";

test("matches snapshot", () => {
  const { asFragment } = render(
    <Offer
      setOffer={() => {}}
      monthlyPayment={377.5}
      newAnnualPercentage={0.142 / 2}
      newMonthlyPayment={282.76}
      newTotalPeriod={84}
    />
  );

  expect(asFragment()).toMatchSnapshot();
});

test("selects an option", async () => {
  const { getByRole } = render(
    <Offer
      setOffer={() => {}}
      monthlyPayment={377.5}
      newAnnualPercentage={0.142 / 2}
      newMonthlyPayment={282.76}
      newTotalPeriod={84}
    />
  );

  const option = getByRole("radio", {
    name: "Lower you payment by 94.74 per month",
  });

  expect(option).toBeInTheDocument();

  expect(option).not.toBeChecked();

  fireEvent.click(option);

  expect(option).toBeChecked();
}, 10000);
