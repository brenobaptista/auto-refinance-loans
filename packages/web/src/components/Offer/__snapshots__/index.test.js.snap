// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="offer"
  >
    <label
      class="loanInfo"
    >
      <span
        class="discount textImportant"
      >
        Lower you payment by 94.74 per month
      </span>
      <input
        class="radio"
        name="offer"
        type="radio"
      />
    </label>
    <div
      class="loanInfo"
    >
      <div>
        APR
      </div>
      <div
        class="textImportant"
      >
        7.10%
      </div>
    </div>
    <hr />
    <div
      class="loanInfo"
    >
      <div>
        Monthly payment
      </div>
      <div
        class="textImportant"
      >
        $282.76
      </div>
    </div>
    <hr />
    <div
      class="loanInfo"
    >
      <div>
        Time remaining
      </div>
      <div
        class="textImportant"
      >
        84 mo
      </div>
    </div>
  </div>
</DocumentFragment>
`;
